docs/writing-rules/data-flow/taint-mode.md

```diff
@@ -182,118 +500,124 @@ pattern-propagators:
   to: $X
 ```
 
-In general, a taint propagator must specify:
-1. A pattern containing **two** metavariables. These two metavariables specify where taint is propagated **from** and **to**.
-2. The `to` and `from` metavariables. These metavariables should match a **variable** or an **expression**.
-    - The `from` metavariable specifies the entry point of the taint.
-    - The `to` metavariable specifies where the tainted data is propagated to, typically an object or data structure.
+### Propagation without side-effect
 
-:::note
-When the `to` metavariable matches a variable (or in general, an l-value), the taint is propagated _by side-effect_.
-See [Sources by side effect](#sources-by-side-effect) for an explanation of what this means.
-:::
+Taint propagators can be used in very imaginative ways, and in some cases we may not want taint to propagate by side-effect. This can be achieved by disabling `by-side-effect` (enabled be default).
```

Comment:
```suggestion
Taint propagators can be used in very imaginative ways, and in some cases you may not want taint to propagate by side-effect. This can be achieved by disabling `by-side-effect`, which is enabled be default.
```